test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DotnetTestHostManagerTests.cs

```diff
@@ -987,6 +987,104 @@ public async Task CleanTestHostAsyncShouldNotThrowIfTestHostIsNotStarted()
 
+    [TestMethod]
+    [DataRow("VSTEST_WINAPPHOST_DOTNET_ROOT(x86)", "DOTNET_ROOT(x86)")]
+    [DataRow("VSTEST_WINAPPHOST_DOTNET_ROOT", "DOTNET_ROOT")]
+    public void ForwardDotnetRootEnvironmentVariableWhenTargetFrameworkIsLessThanNet6SetsCorrectEnvVariable(string envVarName, string expectedForwaredName)
+    {
+        // Arrange
+        const string envVarValue = "c:\\SomePath";
+        _mockEnvironmentVariable.Reset();
+        _mockEnvironmentVariable.Setup(x => x.GetEnvironmentVariable(envVarName)).Returns(envVarValue);
+        string runSettingsXml = """
+            <RunSettings>
+                <RunConfiguration>
+                    <TargetFrameworkVersion>net5.0</TargetFrameworkVersion>
+                </RunConfiguration>
+            </RunSettings>
+            """;
+        _dotnetHostManager.Initialize(_mockMessageLogger.Object, runSettingsXml);
+
+        var startInfo = new TestProcessStartInfo { EnvironmentVariables = new Dictionary<string, string>() };
+        // Sanity check
+        Assert.AreEqual(0, startInfo.EnvironmentVariables.Count);
+
+        // Act
+        _dotnetHostManager.ForwardDotnetRootEnvironmentVariable(startInfo);
+
+        // Assert
+        Assert.AreEqual(1, startInfo.EnvironmentVariables.Count);
+        Assert.IsTrue(startInfo.EnvironmentVariables.TryGetValue(expectedForwaredName, out var actualEnvVarValue));
+        Assert.AreEqual(envVarValue, actualEnvVarValue);
+    }
+
+    [TestMethod]
+    [DataRow("DOTNET_ROOT(x86)", "net5.0")]
+    [DataRow("DOTNET_ROOT(x86)", "net6.0")]
+    [DataRow("DOTNET_ROOT", "net5.0")]
+    [DataRow("DOTNET_ROOT", "net6.0")]
+    [DataRow("DOTNET_ROOT_X86", "net5.0")]
+    [DataRow("DOTNET_ROOT_X86", "net6.0")]
+    [DataRow("DOTNET_ROOT_X64", "net5.0")]
+    [DataRow("DOTNET_ROOT_X64", "net6.0")]
+    public void ForwardDotnetRootEnvironmentVariableWhenIncorrectEnvVarDoesNothing(string envVarName, string framework)
+    {
+        // Arrange
+        const string envVarValue = "c:\\SomePath";
+        _mockEnvironmentVariable.Reset();
+        _mockEnvironmentVariable.Setup(x => x.GetEnvironmentVariable(envVarName)).Returns(envVarValue);
+        string runSettingsXml = $"""
+            <RunSettings>
+                <RunConfiguration>
+                    <TargetFrameworkVersion>{framework}</TargetFrameworkVersion>
+                </RunConfiguration>
+            </RunSettings>
+            """;
+        _dotnetHostManager.Initialize(_mockMessageLogger.Object, runSettingsXml);
+
+        var startInfo = new TestProcessStartInfo { EnvironmentVariables = new Dictionary<string, string>() };
+
+        // Act
+        _dotnetHostManager.ForwardDotnetRootEnvironmentVariable(startInfo);
+
+        // Assert
+        Assert.AreEqual(0, startInfo.EnvironmentVariables.Count);
+    }
+
+    [TestMethod]
+    [DataRow("VSTEST_WINAPPHOST_DOTNET_ROOT(x86)", PlatformArchitecture.X86)]
+    [DataRow("VSTEST_WINAPPHOST_DOTNET_ROOT(x86)", PlatformArchitecture.X64)]
```

Comment:
I am currently not checking if the platform matching makes sense or not and simply forward the first found variable.